getUserFeatureFlag/main.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/Real-Dev-Squad/feature-flag-backend/database"
+	"github.com/Real-Dev-Squad/feature-flag-backend/models"
+	"github.com/Real-Dev-Squad/feature-flag-backend/utils"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+)
+
+func processGetById(userId string, flagId string) (*models.FeatureFlagUserMapping, error) {
+
+	db := database.CreateDynamoDB()
+
+	input := &dynamodb.GetItemInput{
+		TableName: aws.String(database.GetTableName(utils.FFUM_TABLE_NAME)),
+		Key: map[string]*dynamodb.AttributeValue{
+			utils.UserId: {
+				S: aws.String(userId),
+			},
+			utils.FlagId: {
+				S: aws.String(flagId),
+			},
+		},
+	}
+
+	result, err := db.GetItem(input)
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	if result.Item == nil {
+		return nil, nil
+	}
+	featureFlagUserMapping := new(models.FeatureFlagUserMapping)
+	err = dynamodbattribute.UnmarshalMap(result.Item, &featureFlagUserMapping)
+
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+	return featureFlagUserMapping, nil
+}
+
+func handler(req events.APIGatewayProxyRequest) (events.APIGatewayProxyResponse, error) {
```

Comment:
#1 The function could be more robust by handling other possible errors, such as a database error.
#2 The function could be more efficient by caching the results of the processGetById() function.
#3 The function could be more readable by adding comments to explain the logic of the function.

Reply:
Changes for points 1 and 3 are done
Regarding point 2, firstly I think caching is not required for this endpoint of fetching a particular feature flag details based on flag ID and user ID
Additionally, we are still figuring out the efficient way to cache the result and will take it up in a separate PR